client/src/modules/cohort/index.jsx

```diff
@@ -122,8 +128,18 @@ class Cohort extends PureComponent {
     return cohortDetails && cohortDetails.isAdmin;
   };
 
+  handleArchivedListVisibility = id => () => {
+    const { areArchivedListVisible } = this.state;
+    const { fetchArchivedListsMetaData, removeArchivedLists } = this.props;
+    this.setState({ areArchivedListVisible: !areArchivedListVisible });
+    !areArchivedListVisible
+      ? fetchArchivedListsMetaData(id)
+      : removeArchivedLists();
```

Comment:
extract those 3 lines to other function like handleArchivedListsData = () =>
Cause right now this function is not handling only visibility but also data flow management.